src/LaurentMPoly.jl

```diff
@@ -143,24 +143,18 @@ end
 ###############################################################################
 
 @doc raw"""
-    laurent_polynomial_ring(R::Ring, s::Vector{T}; cached::Bool = true) where T <: VarName
-
-Given a base ring `R` and an array of strings `s` specifying how the
-generators (variables) should be printed, return a tuple `T, (x1, x2, ...)`
-representing the new ring $T = R[x1, 1/x1, x2, 1/x2, ...]$ and the generators
-$x1, x2, ...$ of the  ring. By default the parent object `T` will
-depend only on `R` and `x1, x2, ...` and will be cached. Setting the optional
-argument `cached` to `false` will prevent the parent object `T` from being
-cached.
-"""
-function laurent_polynomial_ring(R::Ring, s::AbstractVector{<:VarName}; cached::Bool = true)
-   return Generic.laurent_polynomial_ring(R, [Symbol(v) for v in s], cached=cached)
-end
+    laurent_polynomial_ring(R::Ring, varnames...; cached::Bool = true)
 
-function laurent_polynomial_ring(R::Ring, s::Vector{Symbol}; cached::Bool = true)
-   return Generic.laurent_polynomial_ring(R, s; cached=cached)
-end
+Given a base ring `R` and variable names `varnames...`, say `:x, :y, :z`, return
+a tuple `S, x, y, z` representing the new ring $S = R[x, 1/x, y, 1/y, z, 1/z]$
+and the generators $x, y, z$ of the ring.
 
-function laurent_polynomial_ring(R::Ring, n::Int, s::VarName=:x; cached::Bool = false)
-   return Generic.laurent_polynomial_ring(R, [Symbol(s, i) for i=1:n]; cached=cached)
-end
+By default (`cached=true`), the output `S` will be cached, i.e. if
+`polynomial_ring` is invoked again with the same arguments, the same
+(*identical*) ring is returned. Setting `cached` to `false` ensures a distinct
+new ring is returned, and will also prevent it from being cached.
+
+For ways to give `varnames...` refer to [`polynomial_ring`](@ref) or the
```

Comment:
```suggestion
For ways to specify `varnames...` refer to [`polynomial_ring`](@ref) or the
```
or maybe even
```suggestion
For information about the many ways to specify `varnames...` refer to [`polynomial_ring`](@ref) or the
```

Reply:
Also, `polynomial_ring` doesn't really say much more, does it? And it also has only one example. So maybe just point to `@varnames_interface` and that's it?

(In a follow-up PR we should also added `jldoctest` examples here)

Reply:
There are three `polynomial_ring` docstrings and one for `@polynomial_ring`, but I forgot to include them in the docs.

Reply:
Actually, they all are in the docs and Documenter.jl just links to whatever it thinks it should link to. Interactively, this is fine. Still, we can rearrange the docs later.